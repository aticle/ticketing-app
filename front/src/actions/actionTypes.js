// @flow
export default {
    CREATE_NEW_TICKET: 'CREATE_NEW_TICKET',
    GET_ALL_TICKETS: 'GET_ALL_TICKETS',
    DELETE_TICKET: 'DELETE_TICKET',
    EDIT_TICKET: 'EDIT_TICKET'
}