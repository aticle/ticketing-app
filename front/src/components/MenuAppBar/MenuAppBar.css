.root {
    flex-grow: 1;
}

.menu-btn {
    margin-left: -12px;
    margin-right: 20px;
}