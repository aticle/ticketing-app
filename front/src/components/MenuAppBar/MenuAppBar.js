// @flow
import React, { Component } from 'react';
import { Link, withRouter, RouterHistory } from "react-router-dom";
import { connect, type Dispatch } from 'react-redux';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import Menu from '@material-ui/core/Menu';
import './MenuAppBar.css';
import { logoutUser } from '../../actions/authAction';
import type { State } from '../../reducers';
import type { AuthState } from '../../reducers/authReducer';

type MenuState = {
    anchorEl: EventTarget | null
};
type Props = {
    auth: AuthState,
    history: RouterHistory,
    logoutUser: Dispatch
};

class MenuAppBar extends Component<Props, MenuState> {
    state = {
        anchorEl: null,
    };

    handleClick = (event: Event) => {
        this.setState({ anchorEl: event.currentTarget });
    };

    handleClose = (event: Event) => {
        this.setState({ anchorEl: null });
    };

    logout = (event: Event) => {
        this.handleClose(event);
        this.props.logoutUser(this.props.history);
    }

    render() {
        const { anchorEl } = this.state;
        const { isAuthenticated } = this.props.auth;

        const guestLinks = (
            <div>
                <Link to="/register">
                    <MenuItem
                        onClick={this.handleClose}
                    >Register</MenuItem>
                </Link>
                <Link to="/login">
                    <MenuItem
                        onClick={this.handleClose}
                    >Login</MenuItem>
                </Link>
            </div>
        );

        const authLinks = (
            <div>
                <MenuItem
                    id="logout"
                    onClick={this.logout}
                >Log Out</MenuItem>

                <Link to="/">
                    <MenuItem
                        onClick={this.handleClose}
                    >Tickets List</MenuItem>
                </Link>
                <Link to="/create">
                    <MenuItem
                        onClick={this.handleClose}
                    >Create Ticket</MenuItem>
                </Link>
            </div>
        );

        return (
            <div className="root">
                <AppBar position="static">
                    <Toolbar>
                        <IconButton
                            className="menu-btn"
                            color="inherit"
                            aria-label="Menu"
                            onClick={this.handleClick}
                        >
                            <MenuIcon />
                        </IconButton>
                        <h1>Ticketing App</h1>
                        <Menu
                            id="menu-appbar"
                            anchorEl={anchorEl}
                            anchorOrigin={{
                                vertical: 'top',
                                horizontal: 'right',
                            }}
                            transformOrigin={{
                                vertical: 'top',
                                horizontal: 'right',
                            }}
                            open={Boolean(anchorEl)}
                            onClose={this.handleClose}
                        >
                            {isAuthenticated ? authLinks : guestLinks}
                        </Menu>
                    </Toolbar>
                </AppBar>
            </div>
        );
    }
}

export const mapStateToProps = (state: State) => ({
    auth: state.auth
})

export const mapDispatchToProps = (dispatch: Dispatch) => ({
    logoutUser: (history: RouterHistory) => logoutUser(history)(dispatch)
})

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(MenuAppBar));
