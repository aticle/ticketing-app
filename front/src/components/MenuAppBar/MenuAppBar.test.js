import React from 'react';
import { shallow } from 'enzyme';
import MenuAppBar, { mapDispatchToProps, mapStateToProps } from './MenuAppBar';
import IconButton from '@material-ui/core/IconButton';
import actionTypes from '../../actions/actionTypes';

const props = {
  auth: { isAuthenticated: true, user: {} },
  history: { push: jest.fn },
  logoutUser: jest.fn()
};
const menuAppBar = shallow(<MenuAppBar.WrappedComponent {...props} />);
const menuAppBarI = menuAppBar.instance();

describe('<MenuAppBar />', () => {
  it('renders correctly', () => {
    const props = { auth: { isAuthenticated: false, user: {} } }
    const menuAppBar = shallow(<MenuAppBar.WrappedComponent {...props} />);

    expect(menuAppBar).toMatchSnapshot();
  });

  it('IconButton click updates state', () => {
    menuAppBar.find(IconButton).simulate('click', IconButton);
    expect(menuAppBar.state().anchorEl).not.toEqual(null);
  });

  it('mapStateToProps correctly', () => {
    expect(mapStateToProps(props)).toHaveProperty('auth', props.auth);
  });

  it('mapDispatchToProps correctly', () => {
    const dispatch = jest.fn();

    mapDispatchToProps(dispatch).logoutUser(menuAppBarI.props.history);
    expect(menuAppBarI.props.logoutUser.mock.calls).toEqual([]);
  });

  it('handleClose', () => {
    menuAppBarI.handleClose();
    expect(menuAppBar.state().anchorEl).toEqual(null);
  });
});
