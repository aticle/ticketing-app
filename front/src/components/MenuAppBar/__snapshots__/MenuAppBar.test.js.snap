// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuAppBar /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuAppBar
    auth={
      Object {
        "isAuthenticated": false,
        "user": Object {},
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(AppBar)
        position="static"
      >
        <WithStyles(Toolbar)>
          <WithStyles(IconButton)
            aria-label="Menu"
            className="menu-btn"
            color="inherit"
            onClick={[Function]}
          >
            <pure(MenuIcon) />
          </WithStyles(IconButton)>
          <h1>
            Ticketing App
          </h1>
          <WithStyles(Menu)
            anchorEl={null}
            anchorOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
            id="menu-appbar"
            onClose={[Function]}
            open={false}
            transformOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
          >
            <div>
              <Link
                replace={false}
                to="/register"
              >
                <WithStyles(MenuItem)
                  onClick={[Function]}
                >
                  Register
                </WithStyles(MenuItem)>
              </Link>
              <Link
                replace={false}
                to="/login"
              >
                <WithStyles(MenuItem)
                  onClick={[Function]}
                >
                  Login
                </WithStyles(MenuItem)>
              </Link>
            </div>
          </WithStyles(Menu)>
        </WithStyles(Toolbar)>
      </WithStyles(AppBar)>,
      "className": "root",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Toolbar)>
          <WithStyles(IconButton)
            aria-label="Menu"
            className="menu-btn"
            color="inherit"
            onClick={[Function]}
          >
            <pure(MenuIcon) />
          </WithStyles(IconButton)>
          <h1>
            Ticketing App
          </h1>
          <WithStyles(Menu)
            anchorEl={null}
            anchorOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
            id="menu-appbar"
            onClose={[Function]}
            open={false}
            transformOrigin={
              Object {
                "horizontal": "right",
                "vertical": "top",
              }
            }
          >
            <div>
              <Link
                replace={false}
                to="/register"
              >
                <WithStyles(MenuItem)
                  onClick={[Function]}
                >
                  Register
                </WithStyles(MenuItem)>
              </Link>
              <Link
                replace={false}
                to="/login"
              >
                <WithStyles(MenuItem)
                  onClick={[Function]}
                >
                  Login
                </WithStyles(MenuItem)>
              </Link>
            </div>
          </WithStyles(Menu)>
        </WithStyles(Toolbar)>,
        "position": "static",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(IconButton)
              aria-label="Menu"
              className="menu-btn"
              color="inherit"
              onClick={[Function]}
            >
              <pure(MenuIcon) />
            </WithStyles(IconButton)>,
            <h1>
              Ticketing App
            </h1>,
            <WithStyles(Menu)
              anchorEl={null}
              anchorOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
              id="menu-appbar"
              onClose={[Function]}
              open={false}
              transformOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
            >
              <div>
                <Link
                  replace={false}
                  to="/register"
                >
                  <WithStyles(MenuItem)
                    onClick={[Function]}
                  >
                    Register
                  </WithStyles(MenuItem)>
                </Link>
                <Link
                  replace={false}
                  to="/login"
                >
                  <WithStyles(MenuItem)
                    onClick={[Function]}
                  >
                    Login
                  </WithStyles(MenuItem)>
                </Link>
              </div>
            </WithStyles(Menu)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "aria-label": "Menu",
              "children": <pure(MenuIcon) />,
              "className": "menu-btn",
              "color": "inherit",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Ticketing App",
            },
            "ref": null,
            "rendered": "Ticketing App",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "anchorEl": null,
              "anchorOrigin": Object {
                "horizontal": "right",
                "vertical": "top",
              },
              "children": <div>
                <Link
                  replace={false}
                  to="/register"
                >
                  <WithStyles(MenuItem)
                    onClick={[Function]}
                  >
                    Register
                  </WithStyles(MenuItem)>
                </Link>
                <Link
                  replace={false}
                  to="/login"
                >
                  <WithStyles(MenuItem)
                    onClick={[Function]}
                  >
                    Login
                  </WithStyles(MenuItem)>
                </Link>
              </div>,
              "id": "menu-appbar",
              "onClose": [Function],
              "open": false,
              "transformOrigin": Object {
                "horizontal": "right",
                "vertical": "top",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Link
                    replace={false}
                    to="/register"
                  >
                    <WithStyles(MenuItem)
                      onClick={[Function]}
                    >
                      Register
                    </WithStyles(MenuItem)>
                  </Link>,
                  <Link
                    replace={false}
                    to="/login"
                  >
                    <WithStyles(MenuItem)
                      onClick={[Function]}
                    >
                      Login
                    </WithStyles(MenuItem)>
                  </Link>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <WithStyles(MenuItem)
                      onClick={[Function]}
                    >
                      Register
                    </WithStyles(MenuItem)>,
                    "replace": false,
                    "to": "/register",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Register",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Register",
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <WithStyles(MenuItem)
                      onClick={[Function]}
                    >
                      Login
                    </WithStyles(MenuItem)>,
                    "replace": false,
                    "to": "/login",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Login",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Login",
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(AppBar)
          position="static"
        >
          <WithStyles(Toolbar)>
            <WithStyles(IconButton)
              aria-label="Menu"
              className="menu-btn"
              color="inherit"
              onClick={[Function]}
            >
              <pure(MenuIcon) />
            </WithStyles(IconButton)>
            <h1>
              Ticketing App
            </h1>
            <WithStyles(Menu)
              anchorEl={null}
              anchorOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
              id="menu-appbar"
              onClose={[Function]}
              open={false}
              transformOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
            >
              <div>
                <Link
                  replace={false}
                  to="/register"
                >
                  <WithStyles(MenuItem)
                    onClick={[Function]}
                  >
                    Register
                  </WithStyles(MenuItem)>
                </Link>
                <Link
                  replace={false}
                  to="/login"
                >
                  <WithStyles(MenuItem)
                    onClick={[Function]}
                  >
                    Login
                  </WithStyles(MenuItem)>
                </Link>
              </div>
            </WithStyles(Menu)>
          </WithStyles(Toolbar)>
        </WithStyles(AppBar)>,
        "className": "root",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Toolbar)>
            <WithStyles(IconButton)
              aria-label="Menu"
              className="menu-btn"
              color="inherit"
              onClick={[Function]}
            >
              <pure(MenuIcon) />
            </WithStyles(IconButton)>
            <h1>
              Ticketing App
            </h1>
            <WithStyles(Menu)
              anchorEl={null}
              anchorOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
              id="menu-appbar"
              onClose={[Function]}
              open={false}
              transformOrigin={
                Object {
                  "horizontal": "right",
                  "vertical": "top",
                }
              }
            >
              <div>
                <Link
                  replace={false}
                  to="/register"
                >
                  <WithStyles(MenuItem)
                    onClick={[Function]}
                  >
                    Register
                  </WithStyles(MenuItem)>
                </Link>
                <Link
                  replace={false}
                  to="/login"
                >
                  <WithStyles(MenuItem)
                    onClick={[Function]}
                  >
                    Login
                  </WithStyles(MenuItem)>
                </Link>
              </div>
            </WithStyles(Menu)>
          </WithStyles(Toolbar)>,
          "position": "static",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(IconButton)
                aria-label="Menu"
                className="menu-btn"
                color="inherit"
                onClick={[Function]}
              >
                <pure(MenuIcon) />
              </WithStyles(IconButton)>,
              <h1>
                Ticketing App
              </h1>,
              <WithStyles(Menu)
                anchorEl={null}
                anchorOrigin={
                  Object {
                    "horizontal": "right",
                    "vertical": "top",
                  }
                }
                id="menu-appbar"
                onClose={[Function]}
                open={false}
                transformOrigin={
                  Object {
                    "horizontal": "right",
                    "vertical": "top",
                  }
                }
              >
                <div>
                  <Link
                    replace={false}
                    to="/register"
                  >
                    <WithStyles(MenuItem)
                      onClick={[Function]}
                    >
                      Register
                    </WithStyles(MenuItem)>
                  </Link>
                  <Link
                    replace={false}
                    to="/login"
                  >
                    <WithStyles(MenuItem)
                      onClick={[Function]}
                    >
                      Login
                    </WithStyles(MenuItem)>
                  </Link>
                </div>
              </WithStyles(Menu)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "aria-label": "Menu",
                "children": <pure(MenuIcon) />,
                "className": "menu-btn",
                "color": "inherit",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Ticketing App",
              },
              "ref": null,
              "rendered": "Ticketing App",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "anchorEl": null,
                "anchorOrigin": Object {
                  "horizontal": "right",
                  "vertical": "top",
                },
                "children": <div>
                  <Link
                    replace={false}
                    to="/register"
                  >
                    <WithStyles(MenuItem)
                      onClick={[Function]}
                    >
                      Register
                    </WithStyles(MenuItem)>
                  </Link>
                  <Link
                    replace={false}
                    to="/login"
                  >
                    <WithStyles(MenuItem)
                      onClick={[Function]}
                    >
                      Login
                    </WithStyles(MenuItem)>
                  </Link>
                </div>,
                "id": "menu-appbar",
                "onClose": [Function],
                "open": false,
                "transformOrigin": Object {
                  "horizontal": "right",
                  "vertical": "top",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Link
                      replace={false}
                      to="/register"
                    >
                      <WithStyles(MenuItem)
                        onClick={[Function]}
                      >
                        Register
                      </WithStyles(MenuItem)>
                    </Link>,
                    <Link
                      replace={false}
                      to="/login"
                    >
                      <WithStyles(MenuItem)
                        onClick={[Function]}
                      >
                        Login
                      </WithStyles(MenuItem)>
                    </Link>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <WithStyles(MenuItem)
                        onClick={[Function]}
                      >
                        Register
                      </WithStyles(MenuItem)>,
                      "replace": false,
                      "to": "/register",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Register",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "Register",
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <WithStyles(MenuItem)
                        onClick={[Function]}
                      >
                        Login
                      </WithStyles(MenuItem)>,
                      "replace": false,
                      "to": "/login",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Login",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "Login",
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenThreeAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
